src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferConvertToHexStringOverBitConverter.cs

```diff
@@ -0,0 +1,329 @@
+            /// </summary>
+            /// <param name="invocation">The starting invocation</param>
+            /// <param name="bitConverterToStringInvocation">The extracted BitConverter.ToString invocation, or null if unsuccessful</param>
+            /// <param name="replacementMethod">The Convert.ToHexString* method to use as a replacement</param>
+            /// <param name="toLowerInvocation">The optionally extracted string.ToLower* invocation</param>
+            /// <returns>true if a BitConverter.ToString invocation and its replacement could be extracted</returns>
+            public bool TryGetBitConverterToStringInvocationAndReplacement(
+                IInvocationOperation invocation,
+                [NotNullWhen(true)] out IInvocationOperation? bitConverterToStringInvocation,
+                [NotNullWhen(true)] out IMethodSymbol? replacementMethod,
+                out IInvocationOperation? toLowerInvocation)
+            {
+                bitConverterToStringInvocation = default;
+                replacementMethod = default;
+                toLowerInvocation = default;
+
+                // This check is to prevent two diagnostics when encountering: BitConverter.ToString(data).Replace("-", "").ToLower();
+                // Without this check, this case would report one diagnostic when analyzing Replace and one when analyzing ToLower.
+                if (invocation.Parent is IInvocationOperation parentInvocation &&
+                    IsAnyStringToLowerMethod(parentInvocation.TargetMethod))
+                {
+                    return false;
+                }
+
+                if (IsAnyStringToLowerMethod(invocation.TargetMethod))
+                {
+                    toLowerInvocation = invocation;
+
+                    if (!TryGetInstanceInvocation(invocation, out invocation!))
+                    {
+                        return false;
+                    }
+                }
+
+                if (!IsAnyStringReplaceMethod(invocation.TargetMethod) ||
+                    !HasStringReplaceArgumentsToRemoveHyphen(invocation.Arguments) ||
+                    !TryGetInstanceInvocation(invocation, out invocation!))
+                {
+                    return false;
+                }
+
+                if (IsAnyStringToLowerMethod(invocation.TargetMethod))
+                {
+                    toLowerInvocation = invocation;
+
+                    if (!TryGetInstanceInvocation(invocation, out invocation!))
+                    {
+                        return false;
+                    }
+                }
+
+                // Check if there is a valid replacement that uses Convert.ToHexStringLower or Convert.ToHexString.
+                if (toLowerInvocation is not null &&
+                    TryGetBitConverterReplacementToLower(invocation.TargetMethod, out var replacementToLower))
```

Comment:
```suggestion
                    _bitConverterReplacementsToLower.TryGetValue(invocation.TargetMethod, out var replacementToLower))
```

Reply:
Thanks, replaced it in 03110732e85f9dceebacfa846ce8e4762ab7c2df.